### PR TITLE
Module interface, add the missing method update

### PR DESCRIPTION
**Pull request by gillesbourgeat**

### CODE REVIEW

File: core/lib/Thelia/Module/BaseModuleInterface.php [hunk 1]
```diff
@@ -26,5 +26,7 @@ public function preDeactivation(ConnectionInterface $con = null);
 
     public function postDeactivation(ConnectionInterface $con = null);
 
+    public function update($currentVersion, $newVersion, ConnectionInterface $con = null);
```
Comment (lovenunu) [inline core/lib/Thelia/Module/BaseModuleInterface.php]: What is the goal of that method ?
And what are ```$currentVersion``` and ```$newVersion``` ? Are they Thelia or Module version ? :)

Comment (gillesbourgeat) [inline core/lib/Thelia/Module/BaseModuleInterface.php]: This method already exists, she is just missing in the interface BaseModuleInterface.

https://github.com/gillesbourgeat/thelia/blob/8ccd16d01e68aca05cd0cd9c1180a642006ceb0b/core/lib/Thelia/Module/BaseModule.php#L517-L527

Comment (lovenunu) [inline core/lib/Thelia/Module/BaseModuleInterface.php]: :+1:

**Comment by roadster31** [comment]
By adding the default "null" to $con parameter, you break the compatibility with exiting modules which uses the signature without the default 'null' value, which was valid until now.

File: core/lib/Thelia/Module/BaseModule.php [hunk 2]
```diff
@@ -521,7 +521,7 @@ public function install(ConnectionInterface $con = null)
      * @param string $newVersion the new module version, as defined in the module.xml file
      * @param ConnectionInterface $con
      */
-    public function update($currentVersion, $newVersion, ConnectionInterface $con)
+    public function update($currentVersion, $newVersion, ConnectionInterface $con = null)
```
Comment (roadster31) [inline core/lib/Thelia/Module/BaseModule.php]: By adding the default "null" to $con parameter, you're breaking the compatibility with exiting modules which uses the signature without the default 'null' value, which was valid until now.

Please remove this 'null' default.

Comment (roadster31) [inline core/lib/Thelia/Module/BaseModule.php]: By adding the default "null" to $con parameter, you're breaking the compatibility with exiting modules which uses the signature without the default 'null' value, which was valid until now.

Please remove this 'null' default.

File: core/lib/Thelia/Module/BaseModuleInterface.php (hunk 1 above)
Comment (roadster31) [inline core/lib/Thelia/Module/BaseModuleInterface.php]: Please remove the 'null' default to preserve BC.

**Pull request merged**